### PR TITLE
feat: use structured logging for exceptions

### LINKED ISSUE #275: Use structured logging for the framework logs
The framework logs some events (mostly exceptions), we should use structured logs for them.

### PR DESCRIPTION
**Pull request by coryan**
Log exceptions from the application using structured logging so they are
reported to the right places.

Fixes #275

### CODE REVIEW

File: google/cloud/functions/internal/call_user_function.cc [hunk 1]
```diff
@@ -32,6 +33,33 @@ struct UnwrapResponse {
   }
 };
 
+namespace {
+BeastResponse ApplicationError(nlohmann::json const& error) {
+  auto msg = error.dump();
+  std::cerr << msg << std::endl;
```
Comment (devbww) [inline google/cloud/functions/internal/call_user_function.cc:39]: Debugging remnant?

Comment (coryan) [reply google/cloud/functions/internal/call_user_function.cc:42]: No, but certainly needed a comment explaining why. PTAL.

**Comment by coryan** [comment]
PTAL

File: google/cloud/functions/internal/call_user_function.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/functions/internal/call_user_function.cc:42]: I should have gotten that!  Thanks.

**Review by devbww — APPROVED** [review]

**Pull request merged**